source/common/http/http2/codec_impl.h

```diff
@@ -282,10 +283,12 @@ class ConnectionImpl : public virtual Connection,
     void encodeDataHelper(Buffer::Instance& data, bool end_stream,
                           bool skip_encoding_empty_trailers);
 
+    StreamInfo::BytesMetererSharedPtr& bytesMeterer() override { return bytes_meterer_; }
```

Comment:
nit: I would not return a shared_ptr by reference. Either return a const shared_ptr reference or return a shared_ptr. This can lead to badness pretty easily.

Reply:
Changed to const shared ptr reference.